sycl/include/sycl/accessor.hpp

```diff
@@ -1141,34 +1141,34 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   accessor(const detail::AccessorImplPtr &Impl)
       : detail::AccessorBaseHost{Impl} {}
 
-  id<3> &getOffset() {
+  id<3> getOffset() {
     if constexpr (IsHostBuf)
-      return MAccData->MOffset;
+      return MAccData ? MAccData->MOffset : id<3>();
     else
       return AccessorBaseHost::getOffset();
   }
 
   range<3> &getAccessRange() { return AccessorBaseHost::getAccessRange(); }
```

Comment:
Should we remove the reference on this too?

Reply:
Yes. I know it's not directly related, but it feels odd to keep the interface different for it.

Reply:
I had to delete the reference because we need to return something if MAccData is nullptr to avoid the segmentation fault. For getAccessRange there is no possibility to get segmentation fault as it doesn't accesses MAccData. So, I guess we can leave it as is, if you don't mind.

Reply:
I am okay with leaving it as is, but if it is possible I would like to see a follow-up that removes the non-const reference version, unless we actually need it.

Reply:
[70e5b6f](https://github.com/intel/llvm/pull/8557/commits/70e5b6f6b43a28864d48fcf71846d5a7f7bc0c4f)